python/paddle/python_wrapper_demo/block.py

```diff
@@ -0,0 +1,135 @@
+import paddle.v2.framework.core as core
+
+__all__ = [
+    'Block',
+    'g_block',
+    'block',
+    'eval',
+]
+
+
+class Block(object):
+    '''
+    Block is the concept of code block, which has a sequence of local Variables
+    and Operators.
+    '''
+
+    def __init__(self):
+        '''
+        namespace: str
+        '''
+        self.cmds = []
+
+    def append(self, cmd):
+        '''
+        cmd: Block or Op or Layer
+        '''
+        self.cmds.append(cmd)
+
+    def execute(self):
+        '''
+        Execute this block, this will run all the operators and update the coresponding
+        output variables.
+        '''
+        self._build_nn()
+        self.net.run()
+
+    def _build_nn(self):
+        self.net = core.Net.create()
+        ops = self.__extract_op_from_block(self.cmds)
+        for op in ops:
+            self.net.append_op(op)
+        self.net.complete_add_op(True)
+
+    def __extract_op_from_block(self, cmds):
+        ops = []
+        for cmd in cmds:
+            if type(cmd) is Block:
+                child_ops = self.__extract_op_from_block([cmd])
```

Comment:
Why we must extract op from block?

Reply:
Currently, when a block executes, it will create a NetOp and run it, so operators from multiple block should be extracted and insert into a NetOp. 

This may be changed later to make it more natural.